pkg/serverless/serverless.go

```diff
@@ -223,7 +224,7 @@ func ReportInitError(id ID, errorEnum ErrorEnum) error {
 // WaitForNextInvocation starts waiting and blocking until it receives a request.
 // Note that for now, we only subscribe to INVOKE and SHUTDOWN messages.
 // Write into stopCh to stop the main thread of the running program.
-func WaitForNextInvocation(stopCh chan struct{}, statsdServer *dogstatsd.Server, id ID) error {
+func WaitForNextInvocation(stopCh chan struct{}, statsdServer *dogstatsd.Server, traceAgent *traceAgent.Agent, id ID) error {
```

Comment:
nit: I'm wondering if we should not have the `*Daemon` here instead of the `statsdServer` and `traceAgent`, wdyt?

Reply:
I think it makes sense to refactor the daemon a bit so that it has a shutdown method that deals with the statsDServer and traceAgent. We can probably do that in a follow up PR though.